### ISSUE TITLE
Segfault on comparison of SymEngine objects with incompatible values

### ISSUE THREAD

**Issue by zverok** [symengine/symengine.rb#43]
```ruby
i = SymEngine::Integer.new(5)
# => #<SymEngine::Integer:0x8705938> 
i == 5
# => true 
i == SymEngine::Integer.new(5)
# => true 
i == 'foo'
# [BUG] Segmentation fault at 0x000000
```

**Comment by isuruf** [comment]
Duplicate of #22

**Comment by isuruf** [comment]
Fixed via #53
